### ISSUE TITLE
Connecting to non existent database raises unhandled error event in Windows

### ISSUE THREAD

**Issue by pktippa** [brianc/node-postgres#1217]
```js
var pg = require('pg');
var pool = new pg.Pool({
    "host": "localhost",
    "port": 5432,
    "database": "nonexistentdb",
    "password": "postgres",
    "user": "postgres"
  });
  pool.connect(function(err, client, done) {
      process.nextTick(done);
      console.log("", err);
      console.log("", client);
  });
```
Above sample code try to run in Windows it throws ECONNRESET unhandled error event, where as it doesnt throw in Linux.

**Comment by vitaly-t** [comment]
There is no bug here, this is by design for the driver to throw `error` on the connection when the database or server are not available.

As per the documentation, one must provide the right handler:

```js
pool.on('error', function(err){
    // handler
})
```

**Comment by charmander** [comment]
No, it’s not by design. If the initial connection fails, the error should provided to the `connect` callback and not emitted, which is the behaviour outside of Windows.

**Comment by vitaly-t** [comment]
We've always been required to provide `error` handler when the database connection suddenly dies, i.e. it is unavoidable. And once you have the handler in place, everything works as expected, and you do get the right callback with the right error.

**Comment by charmander** [comment]
Easy to mitigate or not, it’s really not by design.

**Comment by pktippa** [comment]
I do not understand why there is different behavior from linux to windows.

**Comment by vitaly-t** [comment]
> I do not understand why there is different behavior from linux to windows.

For one thing, because this whole driver isn't even tested on Windows, i.e. the tests do not work on Windows.

On a number of occasions I tried to convey this to Brian C that testing it on Windows is paramount, but every time the answer was - I only have a Linux machine here at present. That's it.

I think this is a very sad state of things for such a widely used library. :(

I have tried to make tests work on Windows on 2 occasions, but never succeeded. :(
